container/container_unix.go

```diff
@@ -220,7 +221,9 @@ func (container *Container) UnmountIpcMounts(unmount func(pth string) error) {
 			warnings = append(warnings, err.Error())
 		} else if shmPath != "" {
 			if err := unmount(shmPath); err != nil && !os.IsNotExist(err) {
-				warnings = append(warnings, fmt.Sprintf("failed to umount %s: %v", shmPath, err))
+				if mounted, mErr := mount.Mounted(shmPath); mounted || mErr != nil {
```

Comment:
It's really different since we're only checking if it's mounted if the unmount failed, vs only trying to unmount if it is mounted. Checking if it's mounted first is racey... and I wouldn't be surprised if it's technically slower to do it that way.